gobblin-core-base/src/main/java/gobblin/crypto/KeystoreCredentialStore.java

```diff
@@ -0,0 +1,134 @@
+package gobblin.crypto;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.util.EnumSet;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import lombok.extern.slf4j.Slf4j;
+
+import static gobblin.crypto.KeystoreCredentialStore.CreationOptions.*;
+
+
+@Slf4j
+public class KeystoreCredentialStore implements CredentialStore {
+  private final KeyStore ks;
+  private final char[] password;
+  private final Path path;
+  private final FileSystem fs;
+
+  enum CreationOptions {
+    NONE, CREATE_IF_MISSING
+  };
+
+  public KeystoreCredentialStore(String path, String passwordStr) throws IOException {
+    this(path, passwordStr, EnumSet.noneOf(CreationOptions.class));
+  }
+
+  public KeystoreCredentialStore(String path, String passwordStr, EnumSet<CreationOptions> options) throws IOException {
+    this(FileSystem.get(new Configuration()), new Path(path), passwordStr, options);
```

Comment:
you can also do `new Path(path).getFileSystem(...)`. The advantage of doing this is that the user can specify a file in a different file system.